### ISSUE TITLE
Access your data for all websites permission?

### ISSUE THREAD

**Issue by meglio** [lhall-adexos/xdebug-ext#4]
Dear plugin author,

Can you make it work without the "Access your data for all websites" permission?

-- Thanks, Anton

**Comment by lhall-amphibee** [comment]
Hello Anton,

The plugin needs the permissions to know the browsed URL and place the cookie.

If it's a privacy issue, rest assured that the plugin doesn't share information with anyone (including myself)

**Comment by meglio** [comment]
Yeah I can see this in the source code, but there is still an issue: once I grant the permission, it is there for all future plugin updates. Times goes and the project can get a new maintainer/owner with bad intentions. This actually happens quite often with various extensions, even with very popular ones with 1M+ active installations.

Is there any other way for the plugin to operate and still be useful without accessing the URL and placing the cookie? Something like a middleware that cannot access the full page content, but just alter cookie content. Or add a GET variable to the URL instead, which should work equally well with XDEBUG. What do you think? -- Many thanks

**Comment by lhall-amphibee** [comment]
That's a good point.

I'll try to think of something, but maybe you don't actually need this plugin at all ? Xdebug should work with a GET parameter out of the box with some config : 

**Using a GET parameter**
If you want to debug a script started through a web browser, simply add XDEBUG_SESSION_START=session_name as parameter to the URL. Instead of using a GET parameter, you can also set XDEBUG_SESSION_START as a POST parameter, or through a cookie named XDEBUG_SESSION. Refer to the next section to read on how debug sessions work from within a browser window.

https://2.xdebug.org/docs/remote

Maybe this could work for you ?

**Comment by meglio** [comment]
> Maybe this could work for you ?

Yes and no. While I can definitely add a GET parameter to the URL in my browser, I can't add that to all the API/Ajax calls, dynamic .js and .css file URLs, etc.

So, just a single click to turn xdebug ON/OFF would still be the best experience.
